File: IDE/temp files/newfile153454.cpp
#include <algorithm>
#include <any>
#include <cstdint>
#include <iostream>
#include <optional>
#include <regex>
#include <sstream>
#include <string>
#include <type_traits>
#include <vector>

// Function to escape special characters for regex
std::string escapeRegex(const std::string& str) {
    static const std::regex specialChars{R"([-[\]{}()*+?.,\^$|#\s])"};
    return std::regex_replace(str, specialChars, R"(\$&)");
}
// Function to split a string based on delimiters
std::vector<std::string> LoopParseFunc(const std::string& var, const std::string& delimiter1 = "", const std::string& delimiter2 = "") {
    std::vector<std::string> items;
    if (delimiter1.empty() && delimiter2.empty()) {
        // If no delimiters are provided, return a list of characters
        for (char c : var) {
            items.push_back(std::string(1, c));
        }
    } else {
        // Escape delimiters for regex
        std::string escapedDelimiters = escapeRegex(delimiter1 + delimiter2);
        // Construct the regular expression pattern for splitting the string
        std::string pattern = "[" + escapedDelimiters + "]+";
        std::regex regexPattern(pattern);
        std::sregex_token_iterator iter(var.begin(), var.end(), regexPattern, -1);
        std::sregex_token_iterator end;
        while (iter != end) {
            items.push_back(*iter++);
        }
    }
    return items;
}

// Print function for const char* specifically
void print(const char* value) {
    std::cout << std::string(value) << std::endl;  // Convert const char* to std::string
}
// Print function that converts all types to string if needed
template <typename T>
void print(const T& value) {
    if constexpr (std::is_same_v<T, std::string>) {
        std::cout << value << std::endl;
    } else if constexpr (std::is_same_v<T, char>) {
        std::cout << value << std::endl;
    } else if constexpr (std::is_same_v<T, int>) {
        std::cout << std::to_string(value) << std::endl;
    } else if constexpr (std::is_same_v<T, float>) {
        std::cout << std::to_string(value) << std::endl;
    } else if constexpr (std::is_same_v<T, double>) {
        std::cout << std::to_string(value) << std::endl;
    } else if constexpr (std::is_same_v<T, size_t>) {
        std::cout << std::to_string(value) << std::endl;
    } else if constexpr (std::is_same_v<T, bool>) {
        std::cout << (value ? "1" : "0") << std::endl;
    } else {
        std::cout << "Unsupported type" << std::endl;
    }
}

std::string Trim(const std::string &inputString) {
    if (inputString.empty()) return "";
    size_t start = inputString.find_first_not_of(" \t\n\r\f\v");
    size_t end = inputString.find_last_not_of(" \t\n\r\f\v");
    return (start == std::string::npos) ? "" : inputString.substr(start, end - start + 1);
}

std::string StrReplace(const std::string &originalString, const std::string &find, const std::string &replaceWith) {
    std::string result = originalString;
    size_t pos = 0;
    while ((pos = result.find(find, pos)) != std::string::npos) {
        result.replace(pos, find.length(), replaceWith);
        pos += replaceWith.length();
    }
    return result;
}

int RegExMatch(const std::string& haystack, const std::string& needleRegEx, std::string* outputVar = nullptr, int startingPos = 0) {
    if (haystack.empty() || needleRegEx.empty()) {
        return 0;
    }
    std::regex re(needleRegEx);
    std::smatch match;
    if (std::regex_search(haystack.begin() + startingPos, haystack.end(), match, re)) {
        if (outputVar != nullptr) {
            *outputVar = match.str(0);
        }
        return match.position(0) + 1; // To make it 1-based index
    }
    return 0;
}



std::string parseFunc(std::string expresion) {
    std::string expresionOut = "";
    expresion = Trim(StrReplace(expresion, " ", ""));
    expresion = Trim(StrReplace(expresion, "(", " ( "));
    expresion = Trim(StrReplace(expresion, ")", " ) "));
    expresion = Trim(StrReplace(expresion, "+", " + "));
    expresion = Trim(StrReplace(expresion, "-", " - "));
    expresion = Trim(StrReplace(expresion, "/", " / "));
    expresion = Trim(StrReplace(expresion, "*", " * "));
    expresion = Trim(StrReplace(expresion, ",", " , "));
    std::vector<std::string> expresionForFuncLookInF = "";
    std::vector<std::string> items1 = LoopParseFunc(expresion, " ");
    for (size_t A_Index1 = 0; A_Index1 < items1.size() + 0; A_Index1++) {
        std::string A_LoopField1 = items1[A_Index1 - 0];
        expresionForFuncLookInF.push_back(A_LoopField1);
    }
    expresionForFuncLookInF.push_back("");
    std::vector<std::string> items2 = LoopParseFunc(expresion, " ");
    for (size_t A_Index2 = 0; A_Index2 < items2.size() + 0; A_Index2++) {
        std::string A_LoopField2 = items2[A_Index2 - 0];
        // Check if the variable matches the regex pattern
        if (RegExMatch(A_LoopField2, "^[A-Za-z_][A-Za-z0-9_]*$")) {
            print(A_LoopField2);
        }
    }
    return expresionOut;
}
int main(int argc, char* argv[]) {
    print(parseFunc("-5+func1(bar1, 5+5+(58+5/2--2), func2(VAR1234R, func3(func4(5))))"));
    return 0;
}

File: IDE/temp files/newfile153454.py
import re

def LoopParseFunc(var, delimiter1="", delimiter2=""):
    if not delimiter1 and not delimiter2:
        # If no delimiters are provided, return a list of characters
        items = list(var)
    else:
        # Construct the regular expression pattern for splitting the string
        pattern = r'[' + re.escape(delimiter1) + re.escape(delimiter2) + r']+'
        # Split the string using the constructed pattern
        items = re.split(pattern, var)
    return items

# used the print func

def Trim(inputString):
    return inputString.strip() if inputString else ""

def StrReplace(originalString, find, replaceWith):
    return originalString.replace(find, replaceWith)

def RegExMatch(Haystack, NeedleRegEx, OutputVar=None, StartingPos=0):
    if Haystack is None or NeedleRegEx is None:
        return None
    regex = re.compile(NeedleRegEx)
    match = regex.search(Haystack)
    if match:
        if OutputVar is not None:
            OutputVar.append(match.group(0))
        return match.start() + 1
    else:
        return 0



def parseFunc(expresion):
    expresionOut = ""
    expresion = Trim(StrReplace(expresion, " ", ""))
    expresion = Trim(StrReplace(expresion, "(", " ( "))
    expresion = Trim(StrReplace(expresion, ")", " ) "))
    expresion = Trim(StrReplace(expresion, "+", " + "))
    expresion = Trim(StrReplace(expresion, "-", " - "))
    expresion = Trim(StrReplace(expresion, "/", " / "))
    expresion = Trim(StrReplace(expresion, "*", " * "))
    expresion = Trim(StrReplace(expresion, ",", " , "))
    expresionForFuncLookInF = ""
    items1 = LoopParseFunc(expresion, " ")
    for A_Index1 , A_LoopField1 in enumerate(items1, start=0):
        expresionForFuncLookInF.append(A_LoopField1)
    expresionForFuncLookInF.append("")
    items2 = LoopParseFunc(expresion, " ")
    for A_Index2 , A_LoopField2 in enumerate(items2, start=0):
        # Check if the variable matches the regex pattern
        if (RegExMatch(A_LoopField2, "^[A-Za-z_][A-Za-z0-9_]*$")):
            print(A_LoopField2)
    return expresionOut
print(parseFunc("-5+func1(bar1, 5+5+(58+5/2--2), func2(VAR1234R, func3(func4(5))))"))